aws/aws_secrets_manager/aws_scm.go

```diff
@@ -0,0 +1,220 @@
+package aws_secrets_manager
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/libopenstorage/secrets/aws/utils"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/libopenstorage/secrets"
+	sc "github.com/libopenstorage/secrets/aws/credentials"
+	"github.com/libopenstorage/secrets/pkg/store"
+)
+
+const (
+	// Name of the secret store
+	Name = secrets.TypeAWSSecretsManager
+	// SecretRetentionPeriodInDaysKey is passed through context to the DeleteSecret API
+	SecretRetentionPeriodInDaysKey = "secret-retention-period-in-days"
+)
+
+type awsSecretsMgr struct {
+	creds *credentials.Credentials
+	sess  *session.Session
+	cmk   string
+	asc   sc.AWSCredentials
+	ps    store.PersistenceStore
+	scm   *secretsmanager.SecretsManager
+}
+
+func New(
+	secretConfig map[string]interface{},
+) (secrets.Secrets, error) {
+	if secretConfig == nil {
+		return nil, utils.ErrAWSCredsNotProvided
+	}
+
+	v, _ := secretConfig[utils.AwsRegionKey]
+	region, _ := v.(string)
+	if region == "" {
+		region = os.Getenv(utils.AwsRegionKey)
+		if region == "" {
+			return nil, utils.ErrAWSRegionNotProvided
+		}
+	}
+
+	id, secret, token, err := utils.AuthKeys(secretConfig)
+	if err != nil {
+		return nil, err
+	}
+	asc, err := sc.NewAWSCredentials(id, secret, token, true)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get credentials: %v", err)
```

Comment:
nit: should the error message be more related to failing to form AWS credentials? 
maybe something like "failed to form AWS credentials: %v"

Reply:
makes sense. Will fix the error message